src/node.cc

```diff
@@ -270,6 +272,13 @@ bool v8_initialized = false;
 
 bool linux_at_secure = false;
 
+enum ExceptionOrigin {
+  kFromPromise = true,
+  kFromError = false
+};
+
+static constexpr int kUnhandledPromiseField = 1;
```

Comment:
Can you add a comment explaining that field 0 is used by `PromiseHook()` in async-wrap.cc?  I suspect that won't be immediately obvious to most readers.